examples/cms/src/App.tsx

```
@@ -0,0 +1,134 @@
+import { framer, ManagedCollectionField } from "framer-plugin"
+import "./App.css"
+import { useLayoutEffect, useState } from "react"
+import { DataSource, getDataSources, syncCollection, syncExistingCollection } from "./data"
+import { FieldMapping } from "./FieldMapping"
+import { PLUGIN_KEYS, UI_DEFAULTS } from "./constants"
+import { Spinner } from "./components/Spinner"
+
+const activeCollection = await framer.getManagedCollection()
+
+const syncDataSourceId = await activeCollection.getPluginData(PLUGIN_KEYS.SYNC_REFERENCE)
+const syncSlugFieldId = await activeCollection.getPluginData(PLUGIN_KEYS.SYNC_SLUG)
+
+const result = await syncExistingCollection(activeCollection, syncDataSourceId, syncSlugFieldId)
+
+if (result.status === "success") {
+    await framer.closePlugin(`Synchronization successful`, {
+        variant: "success",
+    })
+}
+
+export function App() {
+    const [isLoadingFields, setIsLoadingFields] = useState(false)
+
+    const [selectedDataSourceId, setSelectedDataSourceId] = useState<string | null>(() => {
+        if (result.status === "needsSetup") {
+            return result.allDataSources[0] || null
+        }
+
+        return syncDataSourceId || null
+    })
+    const [selectedDataSource, setSelectedDataSource] = useState<DataSource | null>(
+        result.status === "needsConfiguration" ? result.dataSource : null
+    )
+    const [existingFields, setExistingFields] = useState<ManagedCollectionField[]>(
+        result.status === "needsConfiguration" ? result.existingFields : []
+    )
+
+    const showFieldsMapping = async (event: React.FormEvent<HTMLFormElement>) => {
+        event.preventDefault()
+
+        if (!selectedDataSourceId) {
+            framer.notify("Please select a data source", { variant: "error" })
+            return
+        }
```

When can this happen?

---

It can't really happen

---

Then I would throw an error instead